packages/jaeger-ui/index.html

```diff
@@ -34,19 +34,15 @@
         const JAEGER_VERSION = DEFAULT_VERSION;
         return JAEGER_VERSION;
       }
+
+      // Workaround some legacy NPM dependencies that assume this is always defined.
+      window.global = {};
+      // Avoid noise from redux-form until https://github.com/redux-form/redux-form/pull/4723 is released.
+      window.module = {};
     </script>
   </head>
   <body>
     <div id="jaeger-ui-root"></div>
-    <!--
-      This HTML file is a template.
-      If you open it directly in the browser, you will see an empty page.
-
-      You can add webfonts, meta tags, or analytics to this file.
-      The build step will place the bundled scripts into the <body> tag.
-
-      To begin the development, run `npm start` in this folder.
-      To create a production bundle, use `npm run build`.
-    -->
+    <script type="module" src="/src/index.tsx"></script>
```

Comment:
It gets replaced with a proper asset at build time (https://vitejs.dev/guide/features.html#typescript).

Reply:
Could you please compare the size of the overall ui-asset bundle after switching to Vite? Add it to the PR description as a benchmarking data point.

Reply:
Looking at the last CI build from `main`, the results from CRA are:
```
 File sizes after gzip:

  863.25 KB  build/static/js/1.2bd86f79.chunk.js
  654.32 KB  build/static/js/main.c8c51bfd.chunk.js
  43.11 KB   build/static/css/1.7f7667b5.chunk.css
  10.62 KB   build/static/css/main.5918f9f5.chunk.css
  763 B      build/static/js/runtime~main.4a686d48.js
```
That'd be ~1519 kB of JS (gzipped) and ~54 kB of CSS (gzipped).

For this branch, we have:
```
 build/index.html                           5.17 kB
build/static/monitor-9000dba4.png        111.77 kB
build/static/jaeger-logo-ab11f618.svg    161.92 kB
build/static/index-b15a508e.css          352.14 kB │ gzip:    56.11 kB
build/static/index-1af2a95e.js         6,008.36 kB │ gzip: 1,642.99 kB
```
So the JS bundle ends up larger. Not immediately sure what the increase is due to; will add `rollup-plugin-visualizer` locally to try and hopefully determine that.

Reply:
One finding: the old build system used `babel-plugin-import` not just for the `antd` handling but also for magically optimizing `lodash` imports. Most places already use `lodash/subComponent` import style to only import needed code, but some import the entire library. Locally, changing those to all use appropriate subcomponent imports shaved off ~30 kB from the gzipped bundle.

With that, the bundle ends up like this:
![Screenshot 2023-03-02 at 20 27 38](https://user-images.githubusercontent.com/2721291/222532867-3a46f709-1003-4d2b-a0ce-aa3d83878c39.png)

It seems the biggest opportunity for saving some bytes would be to lazy-load dependencies that are only needed on specific pages, e.g. `plexus`, or the pyroscope flamegraph visualizer.

Reply:
Indeed, lazy loading those would be good. Does it require changing the build to produce several js files?

Reply:
Looks like this is doable by using dynamic imports for the chunks that we want to lazyload. I tested locally with `DAG.jsx` by moving the top-level imports for `cytoscape` and related deps into the `componentDidMount` callback:

```js
    const [{ default: cytoscape }, { default: cydagre }, { default: dagre }] = await Promise.all([
      import('cytoscape'),
      import('cytoscape-dagre'),
      import('dagre'),
    ]);

```

This split out these dependencies from the main chunk and Vite automatically generated separate chunks for them.

Reply:
For the issue with `lodash`, we could make a separate PR to convert the few usages where the entirety of `lodash` is imported into appropriate sub-component imports. Alternatively, it could also be solved by switching it out for `lodash-es`, which is `lodash` but built as ES modules, making `import { foo } from 'lodash-es'` "just work" without requiring a special plugin (it'd only import `foo` and not the whole library).

Reply:
Filed #1233 and #1234 to track these.